astropy/cosmology/_src/flrw/lambdacdm.py

```diff
@@ -180,11 +180,10 @@ def w(self, z: Quantity | ArrayLike) -> FArray | float:
         redshift z and :math:`\rho(z)` is the density at redshift z, both in
         units where c=1. Here this is :math:`w(z) = -1`.
         """
-        z = aszarr(z)
-        return -1.0 * (np.ones(z.shape) if hasattr(z, "shape") else 1.0)
+        return -1 * np.ones(aszarr(z).shape)
```

Comment:
This does not feel very friendly to different `dtype` or different array classes, somewhat contrary to the arary API spirit. E.g., why should I get a numpy array out if I pass in a z array? Here, I'd suggest,
```
return np.full_like(aszarr(z), -1.0)
```
This puts the burnin on `aszarr` to behave correctly.  (I forget whether `full_like` is supported by the array api; if not, could do `-1 * np.ones_like(asarr(z))`)

Reply:
Actually, why is a full array returned here? Why not just return -1.0 and count on broadcasting? If that has to be the right type (but, really, why???), it could be `- np.ones_like(z, shape=())`.

It seems silly to sacrifice performance just because of type purity!

Reply:
Let me cogitate on that.
It's not type purity that I'm worried about, it's consistency.
1. In general, returning the same-shaped object guarantees things like slicing behaviour.
2. python scalars are different than numpy arrays in terms of the dtype. Part of my Array API goal is to enable preservation of dtypes through computations, e.g. for low and high precision requirements. Ints, obviously can do all that, but in general, floats do not.

Reply:
Sounds good! Definitely worth returning to. If we can keep the behaviours we want and also increase performance that would be best.